system-addon/content-src/components/Card/Card.jsx

```diff
@@ -139,14 +169,15 @@ export class Card extends React.PureComponent {
     const hasImage = link.image || link.hasImage;
     const imageStyle = {backgroundImage: link.image ? `url(${link.image})` : "none"};
 
-    return (<li className={`card-outer${isContextMenuOpen ? " active" : ""}${props.placeholder ? " placeholder" : ""}`}>
-      <a href={link.type === "pocket" ? link.open_url : link.url} onClick={!props.placeholder ? this.onLinkClick : undefined}>
+    return (<li className={`card-outer${isContextMenuOpen ? " active" : ""}${props.placeholder ? " placeholder" : ""}`} onMouseLeave={this.onMouseOut} onMouseEnter={this.onMouseOver}>
+      <a href={link.type === "pocket" ? link.open_url : link.url} onClick={!props.placeholder ? this.onLinkClick : undefined} >
         <div className="card">
           {hasImage && <div className="card-preview-image-outer">
             <div className={`card-preview-image${this.state.imageLoaded ? " loaded" : ""}`} style={imageStyle} />
           </div>}
           <div className={`card-details${hasImage ? "" : " no-image"}`}>
-            {link.hostname && <div className="card-host-name">{link.hostname}</div>}
+            {link.type === "download" && <div className="card-download-icon icon icon-download-folder" />}
+            {link.hostname && <div className="card-host-name">{this.computeHostname()}</div>}
```

Comment:
The Card component has been relatively generic allowing custom behavior via additional optional properties as opposed to type checking.

Alternatively, it looks like the Download Card looks and behaves a bit differently from existing Cards, so maybe it should just be its own Component?

Reply:
@k88hudson any thoughts on if we should care about how generic Card is? It also would potentially affect web extensions that happen to have use a "download" type.. and more generally of special type behaviors.